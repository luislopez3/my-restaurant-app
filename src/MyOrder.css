
button {
  background-color: #f8f0e2;
  color: #56593d;
  margin: auto 50px;
  padding: 5px;
  margin-bottom: 30px;
}
