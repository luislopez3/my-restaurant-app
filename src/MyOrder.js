import "./MyOrder.css";
import React, { useEffect, useState, useContext, useCallback } from "react";
import AppContext from "./AppContext";
import MyOrderForm from "./MyOrderForm";
import OrderItem from "./OrderItem";
import config from "./config";

export default function MyOrder() {
  const [items, setItems] = useState([]);
  const [showForm, setShowForm] = useState(false);
  const { order_id, setOrder } = useContext(AppContext);

  const loadItems = useCallback(
    function () {
      return fetch(`${config.API_URL}/order_items/${order_id}`)
        .then((response) => {
          return response.json();
        })
        .then((data) => {
          setItems(data);
          setOrder({ items: data });
        });
    },
    [setOrder, setItems, order_id]
  );

  useEffect(() => {
    if (!order_id) {
      return;
    }
    fetch(`${config.API_URL}/orders/${order_id}`)
      .then((response) => {
        return response.json();
      })
      .then((data) => {
        loadItems();
      });
  }, [order_id, loadItems]);

  function confirmOrder() {
    return (
      <div>
        <MyOrderForm total_price={calc.value} />
      </div>
    );
  }

  function calculateTotal() {
    let total = 0;
    let tax = 0.05;
    for (let i = 0; i < items.length; i++) {
      total += items[i].price * items[i].quantity;
    }
    return { value: total + total * tax, tax, total };
  }
  const calc = calculateTotal();

  function bagTotal() {
    return (
      <div className="bag-total-row">
        <div className="bag-total-column">
          <h2>Bag Total:</h2>
          <h3>
            Subtotal:{" "}
            {new Intl.NumberFormat("en-US", {
              style: "currency",
              currency: "USD",
            }).format(calc.total)}
          </h3>
          <h3>
            Tax (5%):{" "}
            {new Intl.NumberFormat("en-US", {
              style: "currency",
              currency: "USD",
            }).format(calc.total * calc.tax)}
          </h3>
          <hr />
          <h2>
            Total incl. tax:{" "}
            {new Intl.NumberFormat("en-US", {
              style: "currency",
              currency: "USD",
            }).format(calc.value)}
          </h2>
        </div>
      </div>
    );
  }

  return (
    <>
      {items.map((item, index) => {
        return <OrderItem item={item} index={index} loadItems={loadItems} />;
      })}
      <div className="bag">{bagTotal()}</div>
      <button
        className="confirm-order-button"
        onClick={() => setShowForm(true)}
      >
        Confirm Order
      </button>
      {showForm && confirmOrder()}
    </>
  );
}
