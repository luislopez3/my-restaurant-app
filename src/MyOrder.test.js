import React from "react";
import { create } from "react-test-renderer";
import MyOrder from "./MyOrder";
import { render } from "@testing-library/react";

describe("MyOrder component", () => {
  test("renders MyOrder w/o an error", () => {
    render(<MyOrder />);
  });

  test("Matches the snapshot", () => {
    const wrapper = create(<MyOrder />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
