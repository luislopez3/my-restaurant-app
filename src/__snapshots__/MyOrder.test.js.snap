// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyOrder component Matches the snapshot 1`] = `
Array [
  <div>
    <div>
      <h5>
        Subtotal:
         
        $0.00
      </h5>
      <h5>
        Tax (5%):
         
        $0.00
      </h5>
      <hr />
      <h4>
        Total incl. tax:
         
        $0.00
      </h4>
    </div>
  </div>,
  <button
    onClick={[Function]}
  >
    Confirm Order
  </button>,
]
`;
